src/terminal/forwarding/PortForwardHandler.cpp

```diff
@@ -29,13 +32,45 @@ void PortForwardHandler::update(vector<PortForwardDestinationRequest>* requests,
 }
 
 PortForwardSourceResponse PortForwardHandler::createSource(
-    const PortForwardSourceRequest& pfsr) {
+    const PortForwardSourceRequest& pfsr, string* sourceName) {
   try {
-    auto handler =
-        shared_ptr<PortForwardSourceHandler>(new PortForwardSourceHandler(
-            socketHandler, pfsr.sourceport(), pfsr.destinationport()));
-    sourceHandlers.push_back(handler);
-    return PortForwardSourceResponse();
+    if (pfsr.has_source() && !pfsr.source().has_port()) {
+      throw runtime_error("Do not set a source when forwarding named pipes");
+    }
+    SocketEndpoint source;
+    if (pfsr.has_source()) {
+      source = pfsr.source();
+    } else {
+      // Make a random file to forward the pipe
+      string sourcePattern = string("/tmp/et_forward_sock_XXXXXX");
+      string sourceDirectory = string(mkdtemp(&sourcePattern[0]));
+      ::chmod(sourceDirectory.c_str(), 0777);
```

Comment:
@MisterTea  shouldn't this be 0700?

Reply:
The root user creates the socket and then hands it off to the actual user (check for chmod).

But now that you mention it, it should be possible to have the user process create the socket, I'll look into it.